### DISCUSSION TITLE
`examples` for a `Query` parameter don't render properly in docs

### DISCUSSION THREAD

**Discussion by xoelop** [fastapi/fastapi#8561] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# main.py

from fastapi import FastAPI, Query


app = FastAPI()


@app.get("/")
def test(
    param: str = Query(
        None,
        # example="test",
        examples={"example1": "one example", "example2": "another example"}
    )
):
    return param
```

### Description

Using Python 3.9.6, FastAPI 0.72.0

Run doing `uvicorn main:app --reload --port 8001`
Go to your browser at http://127.0.0.1:8001/docs#/default/test__get or http://127.0.0.1:8001/redoc#/default/test__get

When passing `examples` they aren't rendered properly either in the OpenAPi docs:

<img width="552" alt="Screenshot 2022-01-19 at 14 40 18" src="https://user-images.githubusercontent.com/8508402/150141982-47340d78-e82d-4c47-935e-d615a82bf8d7.png">

This is the output from the console:
```
TypeError: e.get is not a function
    at examples-select.jsx:130:28
    at immutable.js:3016:46
    at immutable.js:2699:43
    at St.__iterate (immutable.js:2206:13)
    at zt.__iterate (immutable.js:2698:25)
    at r.Xt.r.__iterateUncached (immutable.js:3015:23)
    at pe (immutable.js:604:16)
    at r.J.__iterate (immutable.js:274:14)
    at r.forEach (immutable.js:4381:19)
    at immutable.js:2651:16

Uncaught (in promise) TypeError: e.get is not a function
    at examples-select-value-retainer.jsx:194:17
    at immutable.js:3084:23
    at immutable.js:2699:43
    at St.__iterate (immutable.js:2206:13)
    at zt.__iterate (immutable.js:2698:25)
    at r.o.__iterateUncached (immutable.js:3083:16)
    at pe (immutable.js:604:16)
    at r.J.__iterate (immutable.js:274:14)
    at r.forEach (immutable.js:4381:19)
    at immutable.js:2651:16
```

The Redoc docs also don't render it properly
<img width="339" alt="Screenshot 2022-01-19 at 14 40 53" src="https://user-images.githubusercontent.com/8508402/150142053-df6b8a30-c6de-4962-8e31-4c87bf7a89a3.png">

When passing a single `example` it works as expected

### Operating System

macOS

### Operating System Details

2020 Macbook Air (M1)

### FastAPI Version

0.72.0

### Python Version

Python 3.9.6

**Comment by blokje** [comment]
You have to restructure your examples a little bit more elaborate as documented here: https://fastapi.tiangolo.com/tutorial/schema-extra-example/#body-with-multiple-examples

```python
@app.get("/")
def test(
    param: str = Query(
        None,
        examples={
                "example1": {
                        "summary": "First example",
                        "value": "one example",
                },
                "example2": {
                        "summary": "Another example returned when condition X and Y apply",
                        "value": "another example"
                }
        }
    )
):
    return param
```

![image](https://user-images.githubusercontent.com/851418/150818668-622e3254-cc5a-4966-b3ef-9075f9272214.png)

![image](https://user-images.githubusercontent.com/851418/150818728-935a1ddd-37d4-4734-a605-fc7454e4cebb.png)

**Comment by DevinShanahan** [comment]
@tiangolo please let me know if I should open this as a separate issue.
To piggyback on this question, this shows an optional query parameter, but when examples are added, there does not seem to be a way (by default) to select `None` in the UI. My workaround is something like this, but given we know default=None, shouldn't that always be provided as an option? 
```python
@app.get("/")
def test(
    param: str = Query(
        None,
        examples={
                "example1": {
                        "summary": "",
                        "value": None,
                },
                "example2": {
                        "summary": "First example",
                        "value": "one example"
                }
                "example3": {
                        "summary": "Another example returned when condition X and Y apply",
                        "value": "another example"
                }
        }
    )
):
    return param
```

**Comment by BryanEddy5** [comment]
None of these examples are working for me. We are on FastAPI version 0.100.0 and unable to get examples to render on the OpenAPI docs using the examples above:

Has anyone found a working solution?
```
@app.get("/")
def test(
    param: str = Query(
        None,
        examples={
                "example1": {
                        "summary": "",
                        "value": None,
                },
                "example2": {
                        "summary": "First example",
                        "value": "one example"
                }
                "example3": {
                        "summary": "Another example returned when condition X and Y apply",
                        "value": "another example"
                }
        }
    )
):
    return param
````


Edit: 

It appears the issue is that FastAPI puts the `examples` under the schema key while it should be under the `in: query` key.
https://swagger.io/docs/specification/adding-examples/
![image](https://github.com/tiangolo/fastapi/assets/8882754/0ba945a6-81e7-4085-9a2b-b0fcae90f5f5)

**Reply by JonasR** [reply]
I have the same issue. With the recent changes, I don't understand how to define multiple, named examples for a Query parameter.
* Edit: This appears to be a known regression, see here: https://github.com/tiangolo/fastapi/discussions/9794 and the associated PR https://github.com/tiangolo/fastapi/pull/9866
From my understanding there is currently no way to achieve this.

**Comment by tiangolo** [comment, marked as answer]
There's a new parameter `openapi_examples`, to allow declaring examples that show up in Swagger UI while it implements support for multiple JSON Schema examples. It's available now in FastAPI 0.103.0, the new docs are here: https://fastapi.tiangolo.com/tutorial/schema-extra-example/#openapi-specific-examples 🎉
